Allow restricting the creation of resources (#1251)
eclipse/ditto: documentation/src/main/resources/pages/ditto/installation-operating.md

}
```

## Restricting entity creation

By default, Ditto allows anyone to create a new entity (policy or thing) in any namespace. However, this behavior can
be customized, and the ability to create new entities can be restricted.

In the concierge service, you can re-configure the `entity-creation` section to suit your needs. The basic schema is:

```
ditto {
  concierge {
    enforcement {

      # restrict entity creation
      entity-creation {
        grant = [
            {
                resource-types = [],
                namespace-patterns = []
                auth-subjects = []
            }
        ]
        revoke = [
            # same as "grant", but rejecting requests which already passed "grant"
        ]
      }

    }
  }
}
```

When enforcing, the logic is:

* Find a matching entry in the `grant` list
* If one was found, ensure there is no matching entry in the `revoke` list
* If that is the case, accept the request, otherwise deny it

An entry matches, when all of the following conditions are met:

* The resource types list is empty, or contains the requested resource type

> * The namespace patterns list is empty, or contains a pattern that matches the requested namespace

Reviewer: ```suggestion
* The namespaces list is empty, or contains an entry that matches the requested namespace
```

* The auth subjects list is empty, or contains at least one entry of the authorized subjects for the request

This means, an existing entry, with all empty lists, will match. So the default configuration, allowing all access,
can be as simple as:

```
entity-creation {
  grant = [{}]
}
```

The resource types can be any of:

* `policy`
* `thing`

The namespace patterns list, is a list of Java regular expression, which must match the full namespace. This means

> that an expression of `o+` will match `ooo` but not `foo`. Think of it as actually being `^o+$`.

Reviewer: The namespaces are no longer a Java regular expression, but a simple list of namespaces, optionally including a wildcard `*` or `?`.

> The auth subjects list requires only a single entry of the requests auth subjects to match, like `oauth:user-id` or

Reviewer: I wonder if it would be useful to allow the 2 wildcards `*` and `?` in this list as well - what do you think?

That way you could also allow-list all users authenticated via `"keycloak:*"`
